frontend/src/components/common/Icon.vue

```diff
@@ -1,54 +1,33 @@
-<script setup>
-import { computed, defineAsyncComponent } from 'vue';
+<script setup lang="ts">
+import { computed, defineAsyncComponent, type PropType } from 'vue';
+
+enum IconSize {
+    small = '16',
+    medium = '20',
+    large = '24',
+    xLarge = '32'
+}
```

Comment:
You seem to define two enum(s), one here and one at "frontend/src/enums/IconsizeEnum.ts".
I think we can just use the one here, no extra enum file is needed. 
Could you refactor your code for other places? Thanks.

Reply:
I did the refactor. Please let me know if I did in incorrect way